core-bundle/src/Routing/Route404Provider.php

```diff
@@ -56,17 +56,24 @@ public function getRouteCollectionForRequest(Request $request): RouteCollection
 
     public function getRouteByName($name): Route
     {
-        throw new RouteNotFoundException('This router cannot load routes by name');
+        $routes = $this->getRoutes();
+
+        if (!isset($routes[$name])) {
+            throw new RouteNotFoundException('Route name "'.$name.'" is not supported by '.__METHOD__);
+        }
+
+        return $routes[$name];
     }
 
     public function getRoutesByNames($names): array
     {
-        // Support console and web inspector profiling
+        $routes = $this->getRoutes();
+
         if (null === $names) {
-            return $this->getRoutes();
+            return $routes;
         }
 
-        return [];
+        return array_intersect_key($routes, array_flip($names));
```

Comment:
Do we need to cast `(array) $names` here? Because even though the phpDoc says `array|null`, the type is not enforced and could be anything.

Reply:
if the interface would enforce it, you would get an error. Now you get an error as well if you don't follow the docs…